tests/tests/authentication.test.ts

```diff
@@ -0,0 +1,56 @@
+import { authenticationGet, credentialUrl } from './authentication'
+import { TestState } from './test_state'
+import { authorize } from './authorize'
+import { challenge } from './challenge'
+import axios from 'axios'
+import { deriveEncryptionKeyFromSeed } from './utils'
+import * as Kilt from '@kiltprotocol/sdk-js'
+import { toHex } from '@smithy/util-hex-encoding'
+import { CREDENTIAL, DID_KEY_AGREEMENT_URL } from '../test_config'
+import { describe, it, expect } from 'vitest'
+import 'dotenv/config'
+import { EncryptedMessage } from './types'
+
+const mnemonic = process.env.SEED as string
+
+describe('Authentication', () => {
+  it('should refuse unsigned presentation', async () => {
+    const testState = new TestState()
+
+    await authorize(testState)
+    await challenge(testState)
+    await authenticationGet(testState)
+
+    // Use the credential directly without creating a presentation.
+    const payload = {
+      body: { content: [CREDENTIAL], type: 'credential' },
+      createdAt: 0,
+      sender: null,
+      receiver: null,
+      messageId: 0,
+    }
+
+    const seed = Kilt.Utils.Crypto.mnemonicToMiniSecret(mnemonic)
+    const keyAgreementKey = deriveEncryptionKeyFromSeed(seed)
+    const credentialString = JSON.stringify(payload)
+    const encrypted = Kilt.Utils.Crypto.encryptAsymmetric(
+      credentialString,
+      testState.getOpenDidKeyAgreement().publicKey,
+      keyAgreementKey.secretKey,
+    )
+    const postRequestBody: EncryptedMessage = {
+      ciphertext: '0x' + toHex(encrypted.box),
+      senderKeyUri: DID_KEY_AGREEMENT_URL,
+      receiverKeyUri: '-',
+      nonce: '0x' + toHex(encrypted.nonce),
+    }
+
+    const cookie = testState.getCookie()
```

Comment:
This could all be in the setup stage with `vitest` and the testing `it` should contain only the expect and individual test cases where extra functionality needs to be added.

Reply:
well, vitest is more made for unit testing and we're writing end to end testing with everything stateful, I'm not sure how the state of the test can be carried through all the stages of the test. The cookie is used for one flow and it changes on every request. So I'm not sure how this can be done in the setup stage.

Reply:
Ok, that is fair enough.